docs/modules/storage/pages/configuring-persistence.adoc

```diff
@@ -0,0 +1,242 @@
+environments where the members generally use a shared filesystem.
+
+When a member starts, it tries to acquire ownership of the
+first available folder inside the parent folder. If the parent folder
+is empty or if all folders are already acquired by other members, then it creates its own new folder.
+
+=== `persistence.backup-dir`
+
+The folder in which backup snapshots of data
+(hot backups) are stored. See xref:backing-up-persistence.adoc[].
+
+Default: ''
+
+If this element is not defined,
+hot backup is disabled. If a folder is defined which does not exist, it is created on
+the first backup. To avoid clashing data on multiple backups, each backup has a unique
+sequence ID which determines the name of the folder which contains all data for that backup.
+This unique folder is created as a subfolder of the configured `backup-dir`.
+
+=== `persistence.parallelism`
+
+Number of I/O threads that are started concurrently for reading from and writing to Persistence files.
+
+Default: `1`
+
+This is a good default in most but not
+all cases. You should measure the raw I/O throughput of your infrastructure and
+test with different values of parallelism. In some cases such as dedicated
+hardware, higher parallelism can yield more throughput. In other
+cases, such as running on EC2, a higher parallelism can yield diminishing returns with more thread
+scheduling, more contention on I/O, and less efficient garbage collection.
+
+=== `persistence.validation-timeout-seconds`
+
+Validation timeout for the Persistence process
+when validating the cluster members expected to join and the partition table on the whole cluster.
+
+=== `persistence.data-load-timeout-seconds`
+
+Data load timeout for the Persistence process.
+All members in the cluster should finish restoring their local data before this timeout.
+
+=== `persistence.cluster-data-recovery-policy`
+
+Specifies the data recovery policy that is
+respected during the Persistence cluster start.
+
+Default: `FULL_RECOVERY_ONLY`
+
+Valid values are:
+
+* `FULL_RECOVERY_ONLY`: Starts the cluster only when all expected members
+are present and correct. Otherwise, it fails.
+* `PARTIAL_RECOVERY_MOST_RECENT`: Starts the cluster with the members which have most up-to-date partition table and successfully restored their data. All other members leave the cluster and force start themselves. If no member restores its data successfully, cluster start fails.
+* `PARTIAL_RECOVERY_MOST_COMPLETE`: Starts the cluster with the largest group of members which have the same partition table version and successfully restored their data. All other members leave the cluster and force start themselves. If no member restores its data successfully, cluster start fails.
+
+=== `persistence.auto-remove-stale-data`
+
+Enables automatic removal of stale persistence stores.
+When a member terminates or crashes and the cluster state is `ACTIVE`, the remaining
```

Comment:
With 5.0 persistence, a member that crashes while state is  `ACTIVE` will generally not consider its data as stale; it will reload data from disk and then sync data with other members of the cluster as needed. There are a few situations that might result in on-disk data being considered stale, but the conditions under which these occur are exceptional and probably not worth listing exhaustively here. So the option still applies, however the circumstances under which data are considered stale are far less probable to occur in 5.0.

Reply:
If this option is enabled, do members perform checks for stale data, or does the member just automatically remove its persistence store on restart?

Reply:
Regardless of the value of autoRemoveStaleData option, first the rejoining member and cluster's master determine whether data on rejoining member are stale. If stale, then the option's configured value determines whether the rejoining member will automatically delete the data from disk or it will fail startup.